sites/cheerpj/src/content/docs/13-tutorials/05-interoperability-tutorial.mdx

```diff
@@ -0,0 +1,185 @@
+        return "Java received: " + input;
+    }
+
+    public static native void nativeSetApplication(Example myApplication);
+
+    public static void main(String[] args) {
+        Example app = new Example();
+        new Thread(() -> {
+            nativeSetApplication(app);
+            System.out.println("Starting Thread");
+        }).start();
+    }
+}
+```
+
+This Java class contains the following methods:
+
+- `sendToHTML`: A native method implemented in JavaScript, called to send data to the JavaScript environment.
+- `processInput`: Receives data from JavaScript, passes it to `sendToHTML`, and returns a response.
+- `nativeSetApplication`: Captures the running Java thread for persistent communication with JavaScript.
+
+## JavaScript Native Method Implementation
+
+To enable Java and JavaScript communication, we need to implement the [`native`] methods in JavaScript. These methods will handle the data exchange between the two environments.
+
+### 1. Include the CheerpJ Loader Script
+
+Add the following script in the `<head>` section of `index.html`:
+
+```js title="index.html"
+<script src="https://cjrtnc.leaningtech.com/3.0/cj3loader.js"></script>
+```
+
+### 2. Define JavaScript Native Methods
+
+In the `<script>` tag, implement the native methods for interacting with Java.
+
+To implement the native methods in JavaScript, we create an `async` function for each Java method. You can move these methods to a separate `.js` file for better organization and include them using the `<script>` tag in the HTML.
+
+```js title="index.html"
+async function Java_com_example_Example_sendToHTML(lib, str) {
+	document.getElementById("javaOutput").innerText =
+		"JavaScript received: " + str;
+	console.log("Received input from Java: " + str);
+}
+
+async function Java_com_example_Example_nativeSetApplication(
+	lib,
+	myApplication
+) {
+	window.myApplication = myApplication;
+	console.log("Java application instance set on JavaScript side.");
+	return new Promise(() => {}); // Keeps the function from returning
+}
+```
+
+The `Java_com_example_Example_nativeSetApplication` function:
+
+- Assigns the Java application instance to `window.myApplication`.
+- Ensures continuous access by preventing the function from returning.
+
+The `Java_com_example_Example_sendToHTML` function:
+
+- Receives input from Java.
+- Displays the input in the HTML document.
+
+### 3. Initialize CheerpJ
+
+To load CheerpJ, we initialize it in the HTML file with specific configurations for native methods.
+
+```js title="index.html"
+(async () => {
+	await cheerpjInit({
+		version: 8,
+		natives: {
+			Java_com_example_Example_sendToHTML,
+			Java_com_example_Example_nativeSetApplication,
+		},
+	});
+	cheerpjCreateDisplay(800, 600);
+	await cheerpjRunJar("/app/example.jar");
+})();
+```
+
+Explanation:
+
+- `natives`: Lists JavaScript implementations of native Java methods. The naming convention follows `Java_<fully-qualified-class-name>_<method-name>`.
```

Comment:
I would make `cheerpjInit` the bullet point, and explain that `natives` is an init option that you can pass to `cheerpjInit` that is used to implement native methods.